examples/lessons/07/app.js

```diff
@@ -3,205 +3,238 @@
 /* global document */
 
 import {GL, AnimationLoop, Cube, Matrix4, Texture2D,
-  addEvents, loadTextures} from 'luma.gl';
+  addEvents, loadTextures, Model} from 'luma.gl';
+
+const VERTEX_SHADER = `\
+attribute vec3 positions;
+attribute vec2 texCoords;
+attribute vec3 normals;
+
+uniform mat4 uMVMatrix;
+uniform mat4 uPMatrix;
+uniform vec3 uAmbientColor;
+uniform vec3 uLightingDirection;
+uniform vec3 uDirectionalColor;
+uniform bool uUseLighting;
+
+varying vec2 vTextureCoord;
+varying vec3 vLightWeighting;
+
+void main(void) {
+  gl_Position = uPMatrix * uMVMatrix * vec4(positions, 1.0);
+  vTextureCoord = texCoords;
+
+  if (!uUseLighting) {
+      vLightWeighting = vec3(1.0, 1.0, 1.0);
+  } else {
+      // Perform lighting in world space
+      // we should use 'transpose(inverse(mat3(uMVMatrix)))', but
+      // 'inverse' matrix operation not supported in GLSL 1.0
+      mat3 normalMatrix = mat3(uMVMatrix);
```

Comment:
Yeah. We could live with normalMatrix being the same as uMVMatrix as long as the world-to-view transformation doesn't involve non-uniform scaling among different axes.